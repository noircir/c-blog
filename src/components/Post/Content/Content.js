import React from 'react';
import moment from 'moment';
import styles from './Content.module.scss';

const Content = ({ body, title, subtitle, date }) => (
  <article className={styles['content']}>
    <h1 className={styles['content__title']}>{title}</h1>
    <h2 className={styles['content__subtitle']}>{subtitle}</h2>
    <p className={styles['content__date']}><time>{moment(date).format('MMMM DD, YYYY')}</time></p>
    <div className={styles['content__body']} dangerouslySetInnerHTML={{ __html: body }} />
  </article>
);

export default Content;
