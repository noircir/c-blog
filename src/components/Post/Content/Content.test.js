import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  it('renders correctly', () => {
    const props = {
      title: 'test',
      body: '<p>test</p>',
      date: '2019-02-04T12:00:00.000Z',
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
