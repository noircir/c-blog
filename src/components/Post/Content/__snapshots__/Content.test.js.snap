// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<article
  className="content"
>
  <h1
    className="content__title"
  >
    test
  </h1>
  <h2
    className="content__subtitle"
  />
  <p
    className="content__date"
  >
    <time>
      February 04, 2019
    </time>
  </p>
  <div
    className="content__body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</article>
`;
